src/main/java/algorithms/curated170/medium/campusbikes/CampusBikesBucketSort.java

```diff
@@ -0,0 +1,58 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CampusBikesBucketSort {
+
+      int w;
+      List<int[]>[] buckets = new List[2001];
+
+      public int[] assignBikes(int[][] workers, int[][] bikes) {
+            
+            w = workers.length;
+
+            assignToBuckets(workers, bikes);
+
+            return assignFromBuckets();
+      }
+
+      private void assignToBuckets(int[][] workers, int[][] bikes) {
+            for (int i = 0; i < w; i++) {
+                  for (int j = 0; j < w; j++) {
+                        int dis = Math.abs(bikes[j][0] - workers[i][0]) + Math.abs(bikes[j][1] - workers[i][1]);
```

Comment:
In general, your solution is remarkably modular and very nicely organized.

Therefore we can also modularize the distance function as it has no direct dependence on the assignToBucketsFunction in which it will be injected.

```
  private int dist(int[] w, int[] b) {
        return Math.abs(w[0] - b[0]) + Math.abs(w[1] - b[1]);
    }
```